lib/benchee/formatters/csv.ex

```diff
@@ -57,19 +59,30 @@ defmodule Benchee.Formatters.CSV do
       ...> }
       iex> suite
       iex> |> Benchee.Formatters.CSV.format
+      iex> |> Enum.fetch!(0)
       iex> |> (fn({rows, _}) -> Enum.take(rows, 2) end).()
       ["Name,Input,Iterations per Second,Average,Standard Deviation,Standard Deviation Iterations Per Second,Standard Deviation Ratio,Median,Minimum,Maximum,Sample Size\\r\\n",
        "My Job,Some Input,2.0e3,500.0,200.0,800.0,0.4,450.0,200,900,8\\r\\n"]
 
   """
-  @spec format(Suite.t) :: {Enumerable.t, String.t}
+  @spec format(Suite.t) :: [{Enumerable.t, String.t}]
   def format(%Suite{scenarios: scenarios, configuration: configuration}) do
-    rows = scenarios
-           |> Enum.sort_by(fn(scenario) -> scenario.input_name end)
-           |> Enum.map(&to_csv/1)
-           |> add_headers
+    sorted_scenarios = Enum.sort_by(scenarios, fn(scenario) -> scenario.input_name end)
+
+    [
+      {get_general_benchmarks(sorted_scenarios), get_filename(configuration)},
+      {get_raw_benchmarks(sorted_scenarios), @raw_benchmark_output}
```

Comment:
ideally the file name would also be based on the one set in the configuration. The HTML formatter does something similar to this where for instance if you say `benchmark.csv` you will get `benchmark.csv` (main stats) and `benchmark_raw_run_times.csv` i.e. it is interleaved - iirc there is a convenience helper function for that in benchee "core".

Reply:
nice to know. i will use that.